snap/snapcraft.yaml

```
@@ -163,4 +174,24 @@ parts:
         curl -o "${archive}" "https://artifacts.opensearch.org/releases/bundle/opensearch/${version}/${archive}"
         tar -xzvf "${archive}" -C "${CRAFT_PART_INSTALL}/" --strip-components=1
       
+        mkdir -p "${CRAFT_PART_INSTALL}/usr/share/opensearch"
+
+        # OPENSEARCH_HOME points to SNAP_COMMON, whereas the /lib folder sits in /snap
+        sed -i 's@OPENSEARCH_CLASSPATH=\"\$OPENSEARCH_HOME/lib/\*\"@if \[\[ -z "$OPENSEARCH_HOME" \]\]; then OPENSEARCH_CLASSPATH=\"\$OPENSEARCH_LIB\"/*; fi@g' "${CRAFT_PART_INSTALL}"/bin/opensearch-env
+        sed -i 's@bin/opensearch-keystore@\"\${OPENSEARCH_BIN}\"/opensearch-keystore@' "${CRAFT_PART_INSTALL}"/bin/opensearch
+        mv "${CRAFT_PART_INSTALL}/bin" "${CRAFT_PART_INSTALL}/usr/share/opensearch/bin"
+        chmod -R 555 "${CRAFT_PART_INSTALL}/usr/share/opensearch/bin"
+
+        mv "${CRAFT_PART_INSTALL}/jdk" "${CRAFT_PART_INSTALL}/usr/share/opensearch"
+        chmod -R 555 "${CRAFT_PART_INSTALL}/usr/share/opensearch/jdk"
+
+        mv "${CRAFT_PART_INSTALL}/lib" "${CRAFT_PART_INSTALL}/usr/share/opensearch"
+        chmod -R 555 "${CRAFT_PART_INSTALL}/usr/share/opensearch/lib"
+
+        mv "${CRAFT_PART_INSTALL}/modules" "${CRAFT_PART_INSTALL}/usr/share/opensearch/"
+        chmod -R 555 "${CRAFT_PART_INSTALL}/usr/share/opensearch/modules"
+
+        mv "${CRAFT_PART_INSTALL}/plugins" "${CRAFT_PART_INSTALL}/usr/share/opensearch/"
+        chmod -R 555 "${CRAFT_PART_INSTALL}/usr/share/opensearch/plugins"
```

Perhaps with a for loop for readability
```suggestion
        declare -a dirs=(bin jdk lib modules plugins)
        for dir in "${dirs[@]}"; do
            target_dir="${CRAFT_PART_INSTALL}/usr/share/opensearch/${dir}"
            mv "${CRAFT_PART_INSTALL}/${dir}" "${target_dir}"
            chmod -R 555 "${target_dir}"
        done
```